src/main/java/ro/msg/learning/shop/repository/StockRepository.java

```diff
@@ -22,4 +22,7 @@ public interface StockRepository extends JpaRepository<Stock, UUID> {
     @Query(value = "SELECT s FROM Stock s WHERE s.product.id = :uuid AND s.quantity >= :quantity ORDER BY s.quantity DESC LIMIT 1")
     Stock findByProductIdAndQuantityMostAbundant(@Param("uuid") UUID uuid, @Param("quantity") Integer quantity);
 
+
+    @Query(value = "SELECT s FROM Stock s WHERE s.location.id = :uuid")
```

Comment:
There is no need to explicitly write this query by yourself. JPA will know to generate the query behind using the method name. Please read this article for more clarity https://www.baeldung.com/spring-data-derived-queries